patroni/ha.py

```diff
@@ -730,13 +730,19 @@ def _run_cycle(self):
             if self._async_executor.busy:
                 return self.handle_long_action_in_progress()
 
-            # we've got here, so any async action has finished. Check if we tried to recover and failed
+            # we've got here, so any async action has finished.
             if self.recovering and not self.state_handler.need_rewind:
                 self.recovering = False
+                # Check if we tried to recover and failed
                 msg = self.post_recover()
                 if msg is not None:
                     return msg
 
+                if not self.has_lock() and len(self.cluster.members) == 1 and \
+                        not (self.is_synchronous_mode() and self.cluster.sync and
+                             self.cluster.sync.matches(self.state_handler.name)):
+                    return 'too few members in cluster after "recovery", postponing leader race'
+
```

Comment:
Ah, so while previously we allowed a process to wait for the whole loop_wait after calling recover (which was ultimately calling state_handler.follow asynchronously, therefore, had to wait for at least loop_wait afterwards), with this patch the loop will be terminated as soon as the follow will return True, which could be faster than loop_wait (but not necessary), therefore, starting much faster and giving less time for the other cluster members to start. Ultimately, I think this doesn't really give us any guarantees (it could happen that the first 2 nodes to start would be the "slowest" ones) and makes the whole block harder to comprehend, so I vote for removing it.